docs/dev/links.rst

```diff
@@ -0,0 +1,181 @@
+in jdaviz. This linking would then allow e.g. the collapsed dataset to be shown
+as contours on top of the original sliced cube.
+
+This is by far the fastest way of linking, but it does rely on the datasets
+being exactly lined up pixel-wise. This approach can likely be used in specific
+parsers where it is known that the datasets are on the same grid.
+
+Linking by world coordinates
+============================
+
+Using LinkSame
+--------------
+
+There are two main ways of linking by world coordinates. The first is to do
+something similar to how pixel coordinates are linked::
+
+    from glue.core.link_helpers import LinkSame
+
+    world_ids_1 = data1.world_component_ids
+    world_ids_2 = data2.world_component_ids
+
+    for i in range(data1.ndim):
+        data_collection.add_link(LinkSame(world_ids_1[i], world_ids_2[i]))
+
+or see the `following example <https://github.com/spacetelescope/jdaviz/blob/d296c6312b020897034e9dd1fc58c84a2559efa5/jdaviz/app.py#L241-L260>`_
+from jdaviz.
+
+However, this kind of linking is not generally robust because it relies on the
+world coordinates *actually* being the same between the two datasets - so it
+would fail for two images where one image was in equatorial coordinates and the
+other one was in galactic coordinates, because LinkSame would mean that RA was
+the *same* as Galactic longitude, which it is not. Likewise, this would result
+in e.g. wavelength in one dataset being equated with frequency in another. The
+only place this kind of linking could be used is within parsers for specific
+data where it is known with certainty that two world coordinates are the same.
+But in general, wherever possible I think we should phase out any use of
+LinkSame for world coordinates from jdaviz.
+
+Using WCSLink
+-------------
+
+A more robust approach for linking datasets by world coordinates is to use the
+:class:`~glue.plugins.wcs_autolinker.wcs_autolinker.WCSLink` class. Given two
+datasets, and a list of pixel component IDs to link in each dataset, this class
+will set up links between the pixel components by internally representing the
+chain of WCS transformations required. As an example::
+
+    from glue.plugins.wcs_autolinking.wcs_autolinking import WCSLink
+
+    link = WCSLink(data1=data1, data2=data2,
+                cids1=data1.pixel_component_ids, cids2=data2.pixel_component_ids)
+
+    data_collection.add_link(link)
+
+will link all pixel axes between the two datasets, taking into account the WCS
+of ``data1`` and ``data2``.
+
+Note that this should work with any APE 14-compliant WCS, so could link for
+example a FITS WCS to a GWCS instance.
+
+Speeding up WCS links
+=====================
+
+In some cases, doing the full WCS transformations can be slow, and may not be
+necessary if the two datasets are close to each other or overlap significantly.
```

Comment:
Whether or not distortions are present or how much overlap there is doesn't really matter - what matters is more whether the difference between the two datasets can be approximated by an affine transform. You could have two datasets with strong distortions but if the distortions are the same then the datasets are just offset by a translation. I've tried to fix the wording to avoid mentioning overlap and so on.